notebooks/Notebook2.md

```diff
@@ -18,17 +31,49 @@ exciting milestones, preliminary results, etc. Feel free to include images
 code, and any other resources that you think will help clearly convey your
 design process.
 
+I spent a good chunk of this week adapting my Python code into Scala, adding
+some richness to it that takes advantage of Scala's DSL-building features. I
+got pretty stuck while trying to implement a "do for all players" function that
+has a more readable syntax than just using a for loop. The problem is that it's
+very difficult to have the iterator variable accessible in the body of the loop,
+without having an onerous amount of syntax for the user to write.
+
 ## Questions
 
 **What is the most pressing issue for your project? What design decision do
 you need to make, what implementation issue are you trying to solve, or how
 are you evaluating your design and implementation?**
 
+I need to finish implementing Hearts using my new syntax, and see where the
+continuing sticking points in my code are. My Scala code is not currently more
+*efficient* than my Python code, in terms of lines written, but it's already
+significantly more readable. I'm really happy with where my interrupts are going,
+but I need to work out the prompt and eachPlayer functions. In class, I need to
+ask Prof Wiedermann why my deal function is requiring parentheses, and how I
+should lay out my classes- should they all have separate files?
+
 **What questions do you have for your critique partners? How can they best help
 you?**
 
+I'd love to get some advice about implementing prompt and eachPlayer, but I think
+I'll have to ask my critique partners and Prof Wiedermann about them in class, as
+just code reviewing might not necessarily be that helpful. I also want some advice
+about how to lay the classes out in different files, and maybe just pick my partners'
+brains for some more functions that would make sense for my various classes. I also
+definitely need help with some parentheses-free syntax implementation, but again
+I think I'll just have to ask about it in class.
+
 **How much time did you spend on the project this week? If you're working in a
 team, how did you share the work?**
 
+Outside of class, and not including this reflection, I spent around 6-7 hours this week
+working on this project.
+
 **Compared to what you wrote in your contract about what you want to get out of this
 project, how did this week go?**
+
+I think I made it much more progress on design than I expected to, but much less
+progress on implementation than I meant to. I think that's ok, as I expect my
```

Comment:
moood lol. Its hard to make definite project with an open ended project like this. I am struggling alot aswell. we have to remember we only have two more weeks! Scary